packages/amesos2/src/Amesos2_MUMPS.cpp

```diff
@@ -70,10 +71,61 @@ namespace Amesos2 {
   AMESOS2_SOLVER_TPETRA_INST(MUMPS,std::complex<double>,int,int);
```

Comment:
Not sure if putting colons or angle brackets in macro arguments works quite right.  (Spaces, like in `long long`, definitely do _not_ work right.)  Lifting the `TPETRA_ETI_MANGLING_TYPEDEFS()` macro invocation above this code would let you use the typedefs that the macro generates, like `std_complex0double0` and `longlong`.

There's an easier way for you to do this: use Tpetra's instantiation macros.  Look for `packages/tpetra/core/src/TpetraCore_ETIHelperMacros.h` in your _build_ (not source) directory.  Include it in this file, and try this:
```c++
#define AMESOS2_SOLVER_TPETRA_INST_MUMPS( SC, LO, GO ) \
  AMESOS2_SOLVER_TPETRA_INST(MUMPS, SC, LO, GO)

TPETRA_ETI_MANGLING_TYPEDEFS()

TPETRA_INSTANTIATE_SLG(AMESOS2_SOLVER_TPETRA_INST_MUMPS)
```
That would get rid of all of your uses of `AMESOS2_SOLVER_TPETRA_INST`.  You would still need to use `AMESOS2_SOLVER_EPETRA_INST` explicitly, like you do above.

Reply:
Thanks for pointing out the ETI helper macros. I wasn't aware of their existence. This reduced the code to just a few lines in the end. 
I think this macros should be used in all other explicit instantiations too, so new developers will see them easily, when they copy and paste code ;)